backend/eventloop/repocred_reconciler_test.go

```diff
@@ -90,39 +91,123 @@ var _ = Describe("RepoCred Reconcile Function Tests", func() {
 			Expect(repoCredCR.Status.Conditions).To(HaveLen(3))
+			Expect(repoCredCR.Status.Conditions[0].Type).To(Equal(managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred))
+			Expect(repoCredCR.Status.Conditions[0].Reason).To(Equal(managedgitopsv1alpha1.RepositoryCredentialReasonSecretNotFound))
+			Expect(repoCredCR.Status.Conditions[0].Status).To(Equal(metav1.ConditionTrue))
+			Expect(repoCredCR.Status.Conditions[0].Message).To(Equal("Secret specified not found"))
+		})
+
+		It("should set an error status for RepositoryCredentials if the repo and secret exist but the credentials are invalid", func() {
+
+			defer dbq.CloseDatabase()
+
+			By("creating a secret in the cluster")
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-my-repository-credentials-secret",
+					Namespace: apiCRToDatabaseMappingDb.APIResourceNamespace,
+				},
+				Data: map[string][]byte{"username": []byte("username"), "password": []byte("password")},
+			}
+			err := k8sClient.Create(ctx, secret)
+			Expect(err).ToNot(HaveOccurred())
 
+			By("creating a GitOpsDeployment CR in cluster")
+			gitopsDeploymentRepositoryCredentialCR := &managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      apiCRToDatabaseMappingDb.APIResourceName,
+					Namespace: apiCRToDatabaseMappingDb.APIResourceNamespace,
+				},
+				Spec: managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialSpec{
+					Repository: "https://github.com/redhat-appstudio/managed-gitops.git",
+					Secret:     secret.Name,
+				},
+			}
+			err = k8sClient.Create(context.Background(), gitopsDeploymentRepositoryCredentialCR)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("calling the Reconcile function.")
+			reconcileRepositoryCredentials(ctx, dbq, k8sClient, log)
+
+			By("verifing that status is updated for GitopsDeploymentRepositoryCredentialCR.")
+			objectMeta := metav1.ObjectMeta{
+				Name:      apiCRToDatabaseMappingDb.APIResourceName,
+				Namespace: apiCRToDatabaseMappingDb.APIResourceNamespace,
+			}
+			repoCredCR := &managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential{ObjectMeta: objectMeta}
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(repoCredCR), repoCredCR)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(repoCredCR).NotTo(BeNil())
+			Expect(repoCredCR.Status.Conditions).To(HaveLen(3))
+			Expect(repoCredCR.Status.Conditions[0].Type).To(Equal(managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredentialConditionErrorOccurred))
+			Expect(repoCredCR.Status.Conditions[0].Reason).To(Equal(managedgitopsv1alpha1.RepositoryCredentialReasonInvalidCredentials))
+			Expect(repoCredCR.Status.Conditions[0].Status).To(Equal(metav1.ConditionTrue))
+			Expect(repoCredCR.Status.Conditions[0].Message).To(Equal("Repository Credentials provided test-my-repository-credentials-secret for Repository https://github.com/redhat-appstudio/managed-gitops.git are invalid"))
```

Comment:
Minor tweak: as above, rather than expecting the condition to be a position 0, I presume we can search for the condition with the specific Type.

Reply:
Done